### PR TITLE
artical with illustration what is OOP

### PR DESCRIPTION
**Pull request by Mac0490**
artical with illustration what is OOP

### CODE REVIEW

**Comment by liuchengray** [comment]
External feedback:

**negative aspect(s)**:
•	It was not clear 2/12
•	The article was too technical 2/12
•	I lost interest midway 1/12
•	The structure was confusing 2/12
•	I didn't understand the purpose 1/12
**positive aspect(s)**:
•	The information was easy to understand 3/12
•	Good balance between technical depth and readability 1/12
•	I stayed engaged throughout 2/12
•	The structure is clear and easy to follow 3/12
•	I understood the purpose 1/12
**did the drawing help to understand the meaning of the article?**: 42,9 % Yes
**factual errors or inaccuracies in the article**: 
“Do multiple objects form a class? This seems confusing as the objects are instances of the class?”
“When you say that each object has its own attributes and methods, this implies that an object is equivalent to a class or an instance of a class. But, immediately you say that all objects together form a class. This is confusing. It is not clear what a superhero team mean. Does the last line talk about sidekick or the original hero (parent class)?”
**additional comments**:
“As far as I know, Batman can't fly, just drop elegantly.”
“The superhero example seems like a good way to explain the encapsulation. But I think there could be better way to explain inheritance (Or adding some extra context would be also helpful)”

**Comment by erdemkoca** [comment]
### Review Group 2

**Feedback:**

- The revised article is clearer and more engaging, with a good use of the superhero metaphor.
- Removing unnecessary parts has improved readability and conciseness.
- Defining OOP more formally at the start could help beginners before diving into the metaphor.
- Encapsulation and Abstraction could be explained in simpler terms or with additional examples.
- The final sentence is engaging but could tie back to OOP concepts for a stronger conclusion.
- Overall, the article has improved significantly, though minor tweaks could make it even more beginner-friendly.

**Comment by tschudin** [comment]
Here is an alternative story line for consideration, or for a second, contrarian OOP article:

*Object-Oriented Programming*

Computers couldn't care less about "object-oriented programming", but
many programmers are obsessed by it. Their view is that one can break
down the universes into "objects" with well-defined properties and
behavior. Programming then becomes the art of designing and composing
objects - and voila: every problem can be solved, so they believe, by
...
